src/Features/LanguageServer/Protocol/Handler/Definitions/AbstractGoToDefinitionHandler.cs

```diff
@@ -39,9 +39,9 @@ protected async Task<LSP.Location[]> GetDefinitionAsync(LSP.TextDocumentPosition
 
             var position = await document.GetPositionFromLinePositionAsync(ProtocolConversions.PositionToLinePosition(request.Position), cancellationToken).ConfigureAwait(false);
 
-            var definitionService = document.Project.LanguageServices.GetRequiredService<IGoToDefinitionService>();
+            var definitionService = document.Project.LanguageServices.GetRequiredService<IFindDefinitionService>();
```

Comment:
so this doesn't seem like it will work as 3rd parties won't be implementing htis interface . I think you need to probe for *both* interfaces.  If the lang implements IFindDefService, use that, otherwise, fall back.

Reply:
How does this work with the intent of removing the dependency on EditorFeatures? Seems like that would end up with us still depending on that layer here.

Reply:
we'd have to multi-stage this.  Introduce new API, and make Roslyn implement that.  Update this code to use both APIs.  Talk to F#/TS to get them to move to new API (ideally through external-access).  Then update this code to only use the new API.

Yes, this is painful.  But it's something we do a fair amount of for this exact sort of problem.  Fortunately, thanks to our new insertion models, it can be completed in a couple of weeks, instead of months and months :-/

Reply:
Addressed. I've filed https://github.com/dotnet/roslyn/issues/50391 to track this, and introduced the external access API for F# in this PR in a separate commit.